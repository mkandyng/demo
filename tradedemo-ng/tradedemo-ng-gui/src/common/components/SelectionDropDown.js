import React from "react";

export default function SelectionDropDown(props) {
    return (
      <p>
         <label>{props.label}:
              <select id={props.id} name={props.name} onChange={props.handleOnChange}>
                {props.list.map((value) => <option id={props.name+value} value={value}>{value}</option>)}
              </select>
         </label>
     </p>
   )
}
