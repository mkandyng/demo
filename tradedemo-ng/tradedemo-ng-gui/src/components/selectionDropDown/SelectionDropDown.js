import React from "react";

export default function SelectionDropDown(props) {
    const { id,
            label,
            name,
            list,
            handleOnChange } = props;
    return (
      <p>
         <label>{label}:
              <select id={id} name={name} onChange={handleOnChange}>
                {list.map((value) => <option key={name+value} id={name+value} value={value}>{value}</option>)}
              </select>
         </label>
     </p>
   )
}
