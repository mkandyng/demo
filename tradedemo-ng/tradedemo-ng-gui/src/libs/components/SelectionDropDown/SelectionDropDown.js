import React from "react";
import PropTypes from 'prop-types';

/**
 * Drop down with label wrapped with <p>
 */
export default function SelectionDropDown({
  id,
  label,
  name,
  list,
  value,
  handleOnChange
}) {
  return (<p>
    <label>{label}:
      <select id={id} name={name} onChange={handleOnChange} value={value}>
        {list.map((val) => <option key={name + val} id={name + value} value={value}>{val}</option>)}
      </select>
    </label>
  </p>)
}

SelectionDropDown.propTypes = {
  id: PropTypes.string,
  label: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  list: PropTypes.arrayOf(PropTypes.string).isRequired,
  handleOnChange: PropTypes.func.isRequired
};
