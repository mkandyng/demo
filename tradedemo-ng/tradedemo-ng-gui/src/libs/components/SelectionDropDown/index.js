import SelectionDropDown from "./SelectionDropDown";
export default SelectionDropDown;
