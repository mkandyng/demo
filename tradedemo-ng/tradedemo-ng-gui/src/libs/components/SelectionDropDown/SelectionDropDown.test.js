import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import SelectionDropDown from "./SelectionDropDown";

/**
 * This is unit test for the SelectableTabs widget
 * It leverage snapshot to test the presentation
 * and a functional test to check the onSelect call
 */

describe("SelectionDropDown", () => {
  const props = {
    id: "id",
    label: "label",
    name: "name",
    list: [
      "item1", "item2"
    ],
    handleOnChange: jest.fn()
  };

  it("should render component comparing with previous snapshot shallow markup", () => {
    // Given

    // When
    const component = shallow(<SelectionDropDown {...props}/>);

    // Then
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should call onChange prop with given event", () => {
    // Given
    const event = 1
    const component = shallow(<SelectionDropDown {...props}/>);

    // When
    component.find("select").simulate('change', event);

    // Then
    expect(props.handleOnChange).toBeCalledWith(event);
  });

});
