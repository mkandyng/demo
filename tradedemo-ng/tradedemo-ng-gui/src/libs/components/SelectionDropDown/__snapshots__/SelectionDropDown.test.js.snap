// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectionDropDown should render component comparing with previous snapshot shallow markup 1`] = `
<p>
  <label>
    label
    :
    <select
      id="id"
      name="name"
      onChange={[MockFunction]}
    >
      <option
        id="nameundefined"
        key="nameitem1"
      >
        item1
      </option>
      <option
        id="nameundefined"
        key="nameitem2"
      >
        item2
      </option>
    </select>
  </label>
</p>
`;
